launcher/FileSystem.cpp

```diff
@@ -365,14 +365,41 @@ QString getDesktopDir()
     return QStandardPaths::writableLocation(QStandardPaths::DesktopLocation);
 }
 
+bool checkValidShortcutName(QString name)
+{
+#if defined(Q_OS_MACOS)
+    QString invalidOSChars = ":\0";
+#elif defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+    QString invalidOSChars = "/\0";
+#elif defined(Q_OS_WIN)
+    QString invalidOSChars = "<>:\"/\\|?*\0";
+    QString invalidLastWinChars = ". ";
+    if (invalidLastWinChars.contains(name.back()))
+        goto failure; // I am sorry..
+#else
+    return false;
+#endif
+    if (std::any_of(name.begin(), name.end(), [&](QChar const& c){ return invalidOSChars.contains(c); })) {
+        goto failure;
+    }
+    return true;
+failure:
+        qWarning() << "Instance's name contains illegal characters!";
+        return false;
```

Comment:
```suggestion
    bool is_invalid = false;
    QString invalidOSChars;
#if defined(Q_OS_MACOS)
    invalidOSChars = ":\0";
#elif defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
    invalidOSChars = "/\0";
#elif defined(Q_OS_WIN)
    invalidOSChars = "<>:\"/\\|?*\0";
    QString invalidLastWinChars = ". ";
    if (invalidLastWinChars.contains(name.back()))
        is_invalid = true;
#else
    return false;
#endif

    // If is_invalid is already true, short-circuit to avoid needlessly calling the any_of stuff
    is_invalid = is_invalid || std::any_of(name.begin(), name.end(), [&](QChar const& c){ return invalidOSChars.contains(c); });

    if (is_invalid)
        qWarning() << "Instance's name contains illegal characters!";

    return !is_invalid;
```
don't be sorry! :)

Reply:
A great idea!